#20 코멘트 API 작성

Rayoungji opened the pull request:
이슈번호 - 20
작성자 - 라영지
-comment entity/service/repository 생성
-comment controller 생성
> 코멘트 작성 api
> 코멘트 조회 api 
> 코멘트 삭제 api (작성자 인증 후 삭제)
> 가데이터 부분 주석처리 했습니다 (추후 삭제 예정)

Rayoungji committed:
#20 코멘트 API
-comment entity/service/repository 생성
-comment controller 생성
-> 코멘트 작성 api
-> 코멘트 조회 api
-> 코멘트 삭제 api
-> 가데이터 필요한 부분 주석처리 (추후 삭제 예정)

chlalstjd430 commented on youtugo-application/src/main/java/org/nexters/az/comment/controller/CommentController.java:
```diff
@@ -0,0 +1,114 @@
+    @PostMapping("/{postId}/comments/comment")
+    @ResponseStatus(HttpStatus.CREATED)
+    public WriteCommentResponse write(@PathVariable Long postId,
+                                      WriteCommentRequest writeCommentRequest) {
```
@RequestBody 어노테이션이 없는 이유가 있을까요?

chlalstjd430 commented on youtugo-application/src/main/java/org/nexters/az/comment/controller/CommentController.java:
```diff
@@ -0,0 +1,114 @@
+    @ApiOperation("댓글작성")
+    @PostMapping("/{postId}/comments/comment")
+    @ResponseStatus(HttpStatus.CREATED)
+    public WriteCommentResponse write(@PathVariable Long postId,
```
현재 기능은 구현되어있지않지만, @RequestHeader String accessToken을 미리 추가해도 좋을 것 같아요~

chlalstjd430 commented on youtugo-application/src/main/java/org/nexters/az/comment/controller/CommentController.java:
```diff
@@ -0,0 +1,114 @@
+    @ApiOperation("해당 게시글에 대한 댓글들 불러오기")
+    @GetMapping("/{postId}/comments")
+    @ResponseStatus(HttpStatus.OK)
+    public GetCommentsResponse getComments(@PathVariable Long postId) {
```
현재 해당 게시글에 대한 댓글이 1만,100만건일 경우 다 불러오는 구조인데, 적절하게 페이징 처리해서 보여주는게 좋을 것 같아요

chlalstjd430 commented on youtugo-application/src/main/java/org/nexters/az/comment/exception/DeleterNotFoundException.java:
```diff
@@ -0,0 +1,7 @@
+
+import org.nexters.az.exception.UnauthorizedException;
+
+public class DeleterNotFoundException extends UnauthorizedException {
```
댓글 삭제 권한이 없다는 exception이면 클래스명도 적절하게 변경될 필요가 있을 것 같아요,

chlalstjd430 commented on youtugo-application/src/main/java/org/nexters/az/comment/request/WriteCommentRequest.java:
```diff
@@ -0,0 +1,12 @@
+import lombok.Setter;
+
+@Getter
+@Setter
```
Setter가 쓰이는 경우가 있나요?

chlalstjd430 commented on youtugo-domain/src/main/java/org/nexters/az/comment/repository/CommentRepository.java:
```diff
@@ -0,0 +1,13 @@
+import java.util.List;
+
+public interface CommentRepository extends ExtendRepository<Comment> {
+    List<Comment> findAllByPost(Post post);
```
위에 언급한 paging을 위해 Page<Comment> 를 리턴해주고, ByPost보다 ByPostId를 통해 조회하는게 좋을 것 같아요?
아니면 Post객체 자체로 검사하는 이유가 있나요?

chlalstjd430 commented on youtugo-domain/src/main/java/org/nexters/az/comment/repository/CommentRepository.java:
```diff
@@ -0,0 +1,13 @@
+public interface CommentRepository extends ExtendRepository<Comment> {
+    List<Comment> findAllByPost(Post post);
+
+    Comment findByPostAndId(Post post, Long id);
```
위와 마찬가지로 postId로 조회하고, Comment는 Optional로 감싸는게 필요할 것 같아요.

chlalstjd430 reviewed (commented):
리뷰 끝났습니다.
현재 다른 기능들이 없어, 기본적인 테스트 코드를 작성할 수 없는 것을 감안하여 따로 테스트 코드 작성을 이슈로 등록해주시면 감사하겠습니다.
추가적으로 저는 CRUD 한꺼번에 수정해서 코드올리면 읽기 불편할까봐 최대한 이슈 나눠서 진행했는데, 이정도는 크게 불편하지 않네용..저도 나머지 삭제/상세보기 이슈는 합쳐서 진행하도록 해볼게요~

parkyounghwan reviewed (approved):
👍

Rayoungji committed:
Merge branch 'develop' of https://github.com/Nexters/AZ-Server into feature/20
origin devleop 브랜치 풀받음

Rayoungji committed:
#20 코멘트 API 수정
-Controller
  -모든 Controller method에 accessToken 파라미터 추가
  -댓글 조회 api -> 페이징 처리 구현

-DetailedComment
  -Controller의 가독성을 위해 detailedCommentOf / detailedCommentsOf 메소드 옮김

-GetCommentResponse
  -SimplePage 파라미터 추가

-domain/comment/exception
  -댓글 수정 권한 예외처리 클래스 추가

chlalstjd430 commented on youtugo-domain/src/main/java/org/nexters/az/comment/dto/DetailedComment.java:
```diff
@@ -0,0 +1,41 @@
+@Getter
+public class DetailedComment {
+    private Long id;
+    private LocalDateTime created_at;
```
변수명은 camelCase로 표현해주세요~

chlalstjd430 reviewed (approved):
내용 수정후 merge하시면 될 것 같습니다!
수고하셨어요

Rayoungji committed:
#20 코멘트 API 수정
DetailedComment 변수명 수정
